### PR TITLE
Add web-page-visibility

### PR DESCRIPTION
**Pull request by toastal**
[`purescript-web-page-visibility`](https://git.sr.ht/~toastal/purescript-web-page-visibility) is a wrapper for the W3C Page Visibility spec. `v2.0.0` tag [build passed](https://builds.sr.ht/~toastal/job/576410) (Nix + Flakes + Cachix, merely builds as there is not much to test).

[builds.sr.ht status](https://builds.sr.ht/~toastal/purescript-web-page-visibility/commits/.build.yml?)

I have a future plans with this library, but one thing I really wanted to test is the Registry handling non-GitHub git repository—this will be the first repository. Sourcehut’s read-only Git cloning URI is `https://git.sr.ht/ownerCanonicalName/repositoryName` _without_ `.git` (in fact, putting `.git` will return a `403`). As Git is distributed, I will be excited to see what will happen in the future as others see examples of packages on multiple platforms.

Pursuit doc upload though will be blocked by [Pursuit #422](https://github.com/purescript/pursuit/issues/422) and whatever happens there. This may be blocked by #204 which I opened for Sourcehut-specific support or #15 which allows arbitrary git repo support.

### CODE REVIEW

**Comment by f-f** [comment]
Right now the Registry is in a transitional state where we're trying to keep compatibility with Bower as much as we can. Is it possible to install this package with Bower from the SourceHut URL?

**Comment by toastal** [comment]
> install this package with Bower

I'm not entire sure what this means? I _know_ I can't register.

```console
$ bower register purescript-web-page-visibility https://git.sr.ht/~toastal/purescript-web-page-visibility                                                                     web-page-visibility 
bower purescript-web-page-visibility#*          resolve https://git.sr.ht/~toastal/purescript-web-page-visibility#*
bower purescript-web-page-visibility#*         download https://git.sr.ht/~toastal/purescript-web-page-visibility
bower purescript-web-page-visibility#*         resolved https://git.sr.ht/~toastal/purescript-web-page-visibility
? Registering a package will make it installable via the registry (https://registry.bower.io), continue? Yes
bower purescript-web-page-visibility           register https://git.sr.ht/~toastal/purescript-web-page-visibility
bower                                          EUNKNOWN Unknown error: 500 - Registering bower package names is not supported anymore. You can install any bower package on github with command like "bower install jquery/jquery-dist --save"
```

```console
$ bower install https://git.sr.ht/~toastal/purescript-web-page-visibility
bower cached        https://git.sr.ht/~toastal/purescript-web-page-visibility
bower validate      https://git.sr.ht/~toastal/purescript-web-page-visibility#*
bower new           version for https://git.sr.ht/~toastal/purescript-web-page-visibility#*
bower resolve       https://git.sr.ht/~toastal/purescript-web-page-visibility#*
bower download      https://git.sr.ht/~toastal/purescript-web-page-visibility
bower resolved      https://git.sr.ht/~toastal/purescript-web-page-visibility
bower install       purescript-web-page-visibility
```

Downloaded, but left a HTML-rendered `README` as the `index` in `bower_components`

```console
$ bower install https://git.sr.ht/~toastal/purescript-web-page-visibility#v1.0.1
bower ENORESTARGET  URL sources can't resolve targets
```

Tag failed.

To be truthful, I'm by no means a `bower` expert and the docs seem to be screaming to go away at this point. I tried reading some of the `pulp` source to understand but gave up.

**Comment by f-f** [comment]
> I'm not entire sure what this means? I know I can't register

Yeah, I meant running `bower install $repo_url` as you did. Unfortunately it looks like it doesn't work, but I'm no `bower` expert either.
If it turns out that we cannot make it work then I think we cannot add this package as it is - until the Registry is in a transitional state we are committed to keep all packages installable by Bower users.

**Comment by JordanMartinez** [comment]
Per [Bower's docs](https://bower.io/docs/api/#install), does updating the URL to the below work?

`https://git.sr.ht/~toastal/purescript-web-page-visibility/archive/v1.0.4.tar.gz`

**Comment by toastal** [comment]
I could give this a look, but it's not very high on my priorities this second. When I did this, I was using PureScript every day for work and now that I'm not, PureScript is back to hobby status for now.

**Comment by toastal** [comment]
Tried this. Registering is no longer supported. The installing however did work.

File: new-packages.json [hunk 1]
```diff
@@ -184,5 +184,6 @@
   "purescript-unlift": "https://github.com/tweag/purescript-unlift.git",
   "purescript-affjax-node": "https://github.com/purescript-contrib/purescript-affjax-node.git",
   "purescript-affjax-web": "https://github.com/purescript-contrib/purescript-affjax-web.git",
-  "purescript-foreign-readwrite": "https://github.com/artemisSystem/purescript-foreign-readwrite.git"
+  "purescript-foreign-readwrite": "https://github.com/artemisSystem/purescript-foreign-readwrite.git",
+  "purescript-web-page-visibility": "https://git.sr.ht/~toastal/purescript-web-page-visibility"
```
Comment (JordanMartinez) [inline new-packages.json:189]: Is this the `.git` URL? Or just the url to the repo?

Comment (toastal) [reply new-packages.json:189]: > Sourcehut’s read-only Git cloning URI is `https://git.sr.ht/ownerCanonicalName/repositoryName` without `.git` (in fact, putting `.git` will return a 403)

https://github.com/purescript/registry/pull/203#issue-975785057

Comment (JordanMartinez) [reply new-packages.json:189]: Thanks.

**Review by JordanMartinez — APPROVED** [review]

**Comment by toastal** [comment]
I think I noted it elsewhere, but `*.git` is wrong and a 400-level error.
-- 
toastal ไข่ดาว | https://toast.al
PGP: 7944 74b7 d236 dab9 c9ef  e7f9 5cce 6f14 66d4 7c9e

**Pull request merged**